### PR TITLE
Qute - the new Dev UI

### PR DESCRIPTION
**Pull request by mkouba**
Plus some minor fixes and enhancements.

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplateDataBuildItem.java [hunk 1]
```diff
@@ -55,35 +55,36 @@ final class TemplateDataBuildItem extends MultiBuildItem {
         this.properties = propertiesValue != null ? propertiesValue.asBoolean() : false;
     }
 
-    boolean isTargetAnnotatedType() {
-        return targetClass.asClass().name().equals(ValueResolverGenerator.TEMPLATE_DATA);
+    public boolean isTargetAnnotatedType() {
+        AnnotationValue targetValue = annotationInstance.value(ValueResolverGenerator.TARGET);
+        return targetValue == null || targetValue.asClass().name().equals(ValueResolverGenerator.TEMPLATE_DATA);
```
Comment (michalvavrik) [inline extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplateDataBuildItem.java:60]: I kinda expected `targetValue != null`, can you give me a quick hint how is type without that annotation annotation type, please?

Comment (mkouba) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplateDataBuildItem.java:60]: Yes, the `targetValue == null` means that the default value is used (`TemplateData.class` in this particular case). The default values are stored in the annotation defining class, not in the annotation instance. See also the `org.jboss.jandex.AnnotationInstance.valueWithDefault(IndexView, String)`  method and friends.

Comment (mkouba) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplateDataBuildItem.java:60]: And the `TemplateData.class` value means that a value resolver should be generated for the target of this annotation.

Comment (michalvavrik) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplateDataBuildItem.java:60]: Thanks

**Review by phillip-kruger — APPROVED** [review]
LGTM. Not sure about the CI failure

**Comment by mkouba** [comment]
> Depending on timing: #32101 will break this PR. So if this goes in first, I would need to rebase and fix 32101, and if 32101 goes in first, you need to rebase and fix this.

@phillip-kruger pls rebase #32101.

**Pull request merged**